smarts/core/agent_manager.py

```diff
@@ -223,25 +223,27 @@ def observe(
                 else:
                     logging.log(
                         logging.DEBUG,
-                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}",
+                        # pytype: disable=attribute-error
+                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}", 
+                        # pytype: enable=attribute-error
                     )
 
-                rewards[agent_id] = vehicle.trip_meter_sensor(increment=True)
-                scores[agent_id] = vehicle.trip_meter_sensor()
+                rewards[agent_id] = float(vehicle.trip_meter_sensor(increment=True))
+                scores[agent_id] = float(vehicle.trip_meter_sensor())
```

Comment:
I suggest we remove such type casting, i.e., `float(...)` . 
```suggestion
                rewards[agent_id] = vehicle.trip_meter_sensor(increment=True)
                scores[agent_id] = vehicle.trip_meter_sensor()
```

If there is any issue with the `vehicle.trip_meter_sensor()` function, then we should correct it at the source at https://github.com/huawei-noah/SMARTS/blob/60239fbba0c341e331867a38277e480e83c0da85/smarts/core/sensors.py#L938-L942

Reply:
for one thing (and this may not be the problem that `pytype` is complaining about), shouldn't the return type hint be `-> float` for `__call__()`?